### PR TITLE
Fixed Bugs in Pallet Fragments

### PR DESCRIPTION
**Pull request by aang114**
Very short PR! 

 - I will only be able to merge the test cases for pallet fragments after this branch is merged

### CODE REVIEW

**Comment by aang114** [comment]
> Very short PR!
> 
> * I will only be able to merge the test cases for pallet fragments after this branch is merged

only be able to create a PR to merge the test cases for pallet fragments**

File: pallets/fragments/src/lib.rs [hunk 1]
```diff
@@ -677,7 +693,43 @@ pub mod pallet {
 
 			let price = price.saturating_mul(quantity as u128); // `price` = `price` * `quantity`
 
-			// ! Writing if successful
+			if let Some(currency) = fragment_data.metadata.currency { 
+
```
Comment (sinkingsugar) [inline pallets/fragments/src/lib.rs]: less spaces

File: pallets/fragments/src/lib.rs [hunk 2]
```diff
@@ -677,7 +693,43 @@ pub mod pallet {
...
+				let minimum_balance_needed_to_exist = <pallet_balances::Pallet<T> as Currency<T::AccountId>>::minimum_balance();
+
+				let price_balance : <pallet_balances::Pallet<T> as Currency<T::AccountId>>::Balance = price.saturated_into();
+				
+				ensure!(
+					<pallet_balances::Pallet<T> as Currency<T::AccountId>>::free_balance(&who) >= price_balance + minimum_balance_needed_to_exist, 
+					Error::<T>::InsufficientBalance
+				);
+			}
+
```
Comment (sinkingsugar) [inline pallets/fragments/src/lib.rs:721]: relax spaces :)

**Review by sinkingsugar — CHANGES_REQUESTED** [review]

File: pallets/fragments/src/lib.rs (hunk 2 above)
Comment (aang114) [reply pallets/fragments/src/lib.rs:721]: is it better now?

Comment (sinkingsugar) [reply pallets/fragments/src/lib.rs:721]: yep

**Review by sinkingsugar — APPROVED** [review]
LGTM

**Pull request merged**